servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/BuilderUtils.java

```diff
@@ -158,12 +163,27 @@ public static SocketAddress toNettyAddress(Object address) {
             return (SocketAddress) address;
         }
         if (address instanceof HostAndPort) {
-            HostAndPort hostAndPort = (HostAndPort) address;
-            return new InetSocketAddress(hostAndPort.hostName(), hostAndPort.port());
+            return toInetSocketAddress((HostAndPort) address);
         }
         throw new IllegalArgumentException("Unsupported address: " + address);
     }
 
+    /**
+     * Converts {@link HostAndPort} that contains a resolved address into {@link InetSocketAddress}.
+     *
+     * @param resolvedAddress the resolved address to convert.
+     * @return {@link InetSocketAddress} from the passed resolved {@link HostAndPort}.
+     */
+    public static InetSocketAddress toInetSocketAddress(final HostAndPort resolvedAddress) {
```

Comment:
nit: Rename to `toUnresolvedAddress()` to indicate it is unresolved?

Reply:
Actually, it's a resolved address, we expect an IP instead of the host. `toUnresolvedAddress` name is misleading in the same way as `InetSocketAddress.createUnresolved` that doesn't work in this case, because it initializes `InetSocketAddress.getHostName()` instead of `InetSocketAddress.getAddress()`.

Reply:
aah rite .. can we use a name which indicates that it is resolved then?